java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToNestedRecordPattern.java

```diff
@@ -0,0 +1,261 @@
+                    continue;
+                }
+                outer:
+                for (RecordComponentElement recordComponent : type.getRecordComponents()) {
+                    String name = recordComponent.getSimpleName().toString();
+                    String returnType = recordComponent.getAccessor().getReturnType().toString();
+                    returnType = returnType.substring(returnType.lastIndexOf(".") + 1);
+                    if (userVars.get(v.getName().toString()) != null) {
+                        for (UserVariables var : userVars.get(v.getName().toString())) {
+                            if (var.getMethodName().equals(name)) {
+                                bindTree.add((BindingPatternTree) wc.getTreeMaker().BindingPattern(wc.getTreeMaker().Variable(wc.getTreeMaker().
+                                        Modifiers(EnumSet.noneOf(Modifier.class)), var.getVariable(), wc.getTreeMaker().Identifier(returnType), null)));
+                                continue outer;
+                            }
+                        }
+                    }
+                    String baseName = name;
+                    int cnt = 1;
+                    while (SourceVersion.isKeyword(name) || localVars.contains(name)) {
+                        name = CodeStyleUtils.addPrefixSuffix(baseName + cnt++, "", "");
+                    }
+                    localVars.add(name);
+                    bindTree.add((BindingPatternTree) wc.getTreeMaker().BindingPattern(wc.getTreeMaker().Variable(wc.getTreeMaker().
+                            Modifiers(EnumSet.noneOf(Modifier.class)), name, wc.getTreeMaker().Identifier(returnType), null)));
+                }
+                recordComponentMap.put(p, bindTree);
+            }
+
+            DeconstructionPatternTree d = (DeconstructionPatternTree) createNestedPattern((PatternTree) t.getLeaf(), wc, recordComponentMap);
+            while (t != null && t.getLeaf().getKind() != Tree.Kind.IF) {
+                t = t.getParentPath();
+            }
+            IfTree it = (IfTree) t.getLeaf();
+            InstanceOfTree iot = (InstanceOfTree) ((ParenthesizedTree) it.getCondition()).getExpression();
+            StatementTree bt = it.getThenStatement();
+            InstanceOfTree cond = wc.getTreeMaker().InstanceOf(iot.getExpression(), d);
+            List<Tree> removeList = replaceOccurrences.stream().map(tph -> tph.resolve(wc).getLeaf()).collect(Collectors.toList());
+            for(Tree tree : removeList) {
+                bt = wc.getTreeMaker().removeBlockStatement((BlockTree) bt, (StatementTree) tree);
+            }
+            wc.rewrite(it, wc.getTreeMaker().If(wc.getTreeMaker().Parenthesized(cond), bt, it.getElseStatement()));
+        }
+    }
+
+    private static PatternTree createNestedPattern(PatternTree pTree, WorkingCopy wc, Map<PatternTree, List<PatternTree>> map) {
+        if (pTree instanceof BindingPatternTree) {
```

Comment:
Please see the comment about using `instanceof` on the trees.

Reply:
addressed, modified to use switch case